### PR TITLE
Implement configurable connection timeouts on the DefaultHttpProvider

### PR DESCRIPTION
**Pull request by Krisoblucki-okta**
Currently there is no way to configure the connection timeouts to custom values as they are hard coded.  This change allows connection timeouts to be configured.

### CODE REVIEW

**Comment by Krisoblucki-okta** [comment]
@deepak2016 @rgrebski 

Could you guys take a look at this and see if that makes sense.  Ideally we may want to have a IConnectionConfig object encapsulating the timeouts and maybe some other potential connection configuration settings.  What do you think?

File: .gitignore [hunk 1]
```diff
@@ -33,3 +33,9 @@ hs_err_pid*
 /target/
 /pom.xml
 local.properties
+
+# IntelliJ
+.idea
+
+# Output
+/out
```
Comment (NakulSabharwal) [inline .gitignore]: Please revert changes to this file.

**Comment by NakulSabharwal** [comment]
Hi, DefaultConnectionConfig will contain the values of timeout which can be changed using set and get. This config will be passed using getter and setter in DefaultHttpProvider to change value of IConnectionConfig connectionConfig. Timeout and connectionConfig cannot be made final as they can be changed.
The intended usage will be -
`graphClient.getHttpProvider().setConnectionConfig(any class implementing IConnectionConfig);`

**Comment by deepak2016** [comment]
We have fixed the issue using PR: https://github.com/microsoftgraph/msgraph-sdk-java/pull/216
Closing this PR.

**Pull request closed without merging**